api4/file.go

```diff
@@ -325,18 +325,18 @@ func getPublicFile(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if data, err := c.App.ReadFile(info.Path); err != nil {
+	if fileReader, err := c.App.FileReader(info.Path); err != nil {
 		c.Err = err
 		c.Err.StatusCode = http.StatusNotFound
-	} else if err := writeFileResponse(info.Name, info.MimeType, data, true, w, r); err != nil {
+	} else if err := writeFileResponse(info.Name, info.MimeType, fileReader, true, w, r); err != nil {
 		c.Err = err
 		return
 	}
 }
 
-func writeFileResponse(filename string, contentType string, bytes []byte, forceDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
+func writeFileResponse(filename string, contentType string, fileReader io.Reader, forceDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
 	w.Header().Set("Cache-Control", "max-age=2592000, private")
-	w.Header().Set("Content-Length", strconv.Itoa(len(bytes)))
+	w.Header().Set("Transfer-Encoding", "chunked")
```

Comment:
I think we can still send the "Content-Length" header for the main files because we store the file size in the file info here: https://github.com/mattermost/mattermost-server/blob/master/model/file_info.go#L30
I don't think we store the size of the preview images. Maybe we should?

Reply:
I am just thinking about the download indicators on browsers, if we don't send the `Content-Length` then they won't show progress.

Reply:
@crspeller Yeah, the preview/thumbnail image sizes do not exist. And I think it could be a migration issue.
How about using chunked response only for preview/thumbnail API?

Reply:
I don't think the preview and thumbnail images are not for file downloading, and not so large anyway. It looks ok.